src/coreclr/jit/emitarm64.cpp

```diff
@@ -7684,6 +7684,12 @@ void emitter::emitIns_R_S(instruction ins, emitAttr attr, regNumber reg1, int va
     // Try to optimize a load/store with an alternative instruction.
     if (isLdrStr && emitComp->opts.OptimizationEnabled() && OptimizeLdrStr(ins, attr, reg1, reg2, imm, size, fmt))
     {
+        // Ensure local variables are tracked.
+        emitLastIns->idAddr()->iiaLclVar.initLclVarAddr(varx, offs);
```

Comment:
I am wondering `emitLastIns` would already have this set while processing the previous `ldr`, right?

Reply:
For the `IsRedundantLdStr` case, yes, I think it's essentially the same information in both instructions.

Reply:
Correct, so wondering how this would help? What extra information would we track with this that was not already present in `emitLastIns`?

Reply:
`OptimizeLdrStr()` is attempting both `IsRedundantLdStr()` and `ReplaceLdrStrWithPairInstr()`.

For the `IsRedundantLdStr()` it's redundant.

For the `ReplaceLdrStrWithPairInstr()` case (where we are merging two `str`s into an `stp`), then the local variable in the first str has already been tracked, but the local variable in the second str needs tracking.

Reply:
That makes sense.